pkg/vault/vault.go

```diff
@@ -42,6 +95,10 @@ func GetSecrets(ctx context.Context, secretValues *map[string]map[string]string,
 	auth := NewAuthClient()
```

Comment:
I think here is where we should be determining the type of auth client to use.  Originally this was only setup with the gcp auth client, so that's what NewAuthClient returned, however we should rename that original auth client to NewGcpAuthClient, and create a NewGitHubAuthClient.  Both auth clients satisfy the existing auth client interface (GetVaultToken) and by adding that code here, and passing it into NewVaultClient, we don't need to modify as much of the code as we are right now.

I think it would be a bit of undoing some of the combined code you have, but it would make use of the interface and really reduce the overall complexity of adding this new auth client.

Reply:
In the end we would keep this function
```go
func (a *gcpAuthClient) GetVaultToken(vc *vaultClient) (string, error) {}
```

rename this one from 

```go
// NewAuthClient returns a new instance of an auth client
func NewAuthClient() AuthClient {
	a := &gcpAuthClient{}
	return a
}
```

to

```go
// NewAuthClient returns a new instance of an auth client
func NewGcpAuthClient() AuthClient {
	a := &gcpAuthClient{}
	return a
}
```

and add a new type and function to support the github vault client

```go
type githubAuthClient struct {
}
// NewAuthClient returns a new instance of an auth client
func NewGithubAuthClient() AuthClient {
	a := &githubAuthClient{}
	return a
}
func (a *githubAuthClient) GetVaultToken(vc *vaultClient) (string, error) {}
```

Reply:
Each auth client handles its handshake with vault in its own way, but the interface is satisfied, and the both take a vault client, and return a token or an error